pkg/sidecar/param/pub_api.go

```diff
@@ -0,0 +1,238 @@
+// Copyright © 2018 One Concern
+package param
+
+import (
+	"errors"
+)
+
+type fuseParamsBundleParams struct {
+	Name         string `json:"name" yaml:"name"`
+	SrcPath      string `json:"srcPath" yaml:"srcPath"`
+	SrcRepo      string `json:"srcRepo" yaml:"srcRepo"`
+	SrcLabel     string `json:"srcLabel" yaml:"srcLabel"`
+	SrcBundle    string `json:"srcBundle" yaml:"srcBundle"`
+	DestPath     string `json:"destPath" yaml:"destPath"`
+	DestRepo     string `json:"destRepo" yaml:"destRepo"`
+	DestMessage  string `json:"destMessage" yaml:"destMessage"`
+	DestLabel    string `json:"destLabel" yaml:"destLabel"`
+	DestBundleID string `json:"destBundleID" yaml:"destBundleID"`
+	_            struct{}
+}
+
+type FUSEParams struct {
+	Globals struct {
+		SleepInsteadOfExit bool   `json:"sleepInsteadOfExit" yaml:"sleepInsteadOfExit"`
+		CoordPoint         string `json:"coordPoint" yaml:"coordPoint"`
+		Contributor        struct {
+			Name  string `json:"name" yaml:"name"`
+			Email string `json:"email" yaml:"email"`
+			_     struct{}
+		} `json:"contributor" yaml:"contributor"`
+		_ struct{}
+	} `json:"globalOpts" yaml:"globalOpts"`
+	Bundles []fuseParamsBundleParams `json:"bundles" yaml:"bundles"`
+	_       struct{}
+}
+
+type pgParamsDBParams struct {
+	Name        string `json:"name" yaml:"name"`
+	Port        int    `json:"pgPort" yaml:"pgPort"`
+	DestRepo    string `json:"destRepo" yaml:"destRepo"`
+	DestMessage string `json:"destMessage" yaml:"destMessage"`
+	DestLabel   string `json:"destLabel" yaml:"destLabel"`
+	SrcRepo     string `json:"srcRepo" yaml:"srcRepo"`
+	SrcLabel    string `json:"srcLabel" yaml:"srcLabel"`
+	SrcBundle   string `json:"srcBundle" yaml:"srcBundle"`
```

Comment:
✨ 

yes, i didn't include it b/c i've considered the specificity of the bundle id file as useful albeit not useful enough.

i'd prefer doing away with the specificity of recording bundle ids to individual files store a single serialization format rather than lots of individual wires.

the 👍 of this towards consistency.